classes/Cart.php

```diff
@@ -1819,9 +1814,9 @@ public function getPackageList($flush = false)
 			$id_warehouse = 0;
 			foreach ($warehouse_count_by_address[$product['id_address_delivery']] as $id_war => $val)
 			{
-				if (in_array((int)$id_war, $product['warehouse_list']))
+				if (array_key_exists((int)$id_war, $product['warehouse_list']))
 				{
-					$product['carrier_list'] = array_merge($product['carrier_list'], Carrier::getAvailableCarrierList(new Product($product['id_product']), $id_war, $product['id_address_delivery'], null, $this));
+					$product['carrier_list'] = array_replace($product['carrier_list'], Carrier::getAvailableCarrierList(new Product($product['id_product']), $id_war, $product['id_address_delivery'], null, $this));
```

Comment:
@guestisp,

We are not the only one to do so out there:
https://wordpress.org/about/requirements/

I think you would be surprised to see how many people/hosting providers still use this PHP version.

Anyway we are upgrading to PHP 5.4 as of 1.6.1.0 release.

Don't forget to have a look at build.prestashop.com. This is were we write about PrestaShop future :)